spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -313,7 +313,7 @@ private SparkTableUtil() {
     }
   }
 
-  private static List<DataFile> listParquetPartition(Map<String, String> partitionPath, String partitionUri,
+  private static List<DataFile> listParquetPartition(Map<String, String> partitionPath, URI partitionUri,
                                                      PartitionSpec spec, Configuration conf,
                                                      MetricsConfig metricsSpec) {
     try {
```

Comment:
The alternate fix is on the line below (an all other times we make a Path from a string), where instead of just passing through the string we pass back through the URI version of the string, or possibly just decode the string before passing it through. I think it's probably safest to use URI the whole time, second safest to change back to URI at the last moment, third safest to attempt to decode and hope Hadoop parses the string like we want it too.